### PR TITLE
feat: Add modal notification

### LINKED ISSUE #3346: [NEW FEATURE] - Show error modal when making action with wrong timestamp
### Briefly describe the feature.

Add error message upon joining space with wrong timestamp

### Which problem is this feature trying to solve?

It's a recurring support issue that users with unsynchronized device clock can't join spaces and there in no error for this atm.

### What is the expected solution?

Adding an appropriate error, possible better to show an error in layman terms like "Your computer time is not accurate" instead of Oops wrong timestamp that doesn't tell much to the average user. It would also help if the error not collapse after a couple of second and show a link to the relevant solution:
https://github.com/snapshot-labs/snapshot/discussions/769

### How do you currently handle this problem?

Currently it cause frustration to users and eventually a small part of them can solve it after unnecessary use of the support. Assuming for each user that solve the issue 10 may give up on Snapshot.

### Anything else you'd like to mention?

_No response_

### PR DESCRIPTION
**Pull request by samuveth**
Fixes #3346

We already have flash notifications with `useFlashNofication.ts` but sometimes it's useful to give the user more information. For this I've created `useModalNotification.ts` which works very similar just with a modal.

The first use case for this is the `wrong timestamp` error when signing any message (voting, joining space, creating proposal). With this information the user should be able to solve the issue.

image

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -48,5 +48,6 @@ onMounted(async () => init());
       </div>
     </div>
   </template>
-  <BaseFlashNotification />
+  <TheFlashNotification />
+  <TheModalNotification />
```
Comment (ChaituVR) [inline src/App.vue:52]: Why `The` keyword 😄 any reasons?

File: src/components/ModalVote.vue [hunk 2]
```diff
@@ -86,8 +86,8 @@ async function handleSubmit() {
     emit('openPostVoteModal');
     emit('reload');
     addVotedProposalId(props.proposal.id);
-    emit('close');
   }
+  emit('close');
```
Comment (ChaituVR) [inline src/components/ModalVote.vue:90]: Will close the vote modal even if it is not a success?

File: src/App.vue (hunk 1 above)
Comment (samuveth) [reply src/App.vue:52]: @ChaituVR https://vuejs.org/style-guide/rules-strongly-recommended.html#single-instance-component-names

File: src/components/ModalVote.vue (hunk 2 above)
Comment (samuveth) [reply src/components/ModalVote.vue:90]: Yes, this is because we want to show the other modal. The way the modal component is setup it will not look good if they are stacked with different heights.

File: src/locales/default.json [hunk 3]
```diff
@@ -282,6 +282,12 @@
     "all": "All",
     "unread": "Unread"
   },
+  "modalNotifications": {
+    "wrong timestamp": {
+      "title": "Wrong timestamp",
+      "message": "The message timestamp is not valid. Please check your devices clock and make sure it is set to the correct date and time."
```
[the remark this answers is not in the document]
Comment (samuveth) [reply src/locales/default.json]: Much better, thanks!

File: src/components/ModalVote.vue (hunk 2 above)
Comment (ChaituVR) [reply src/components/ModalVote.vue:90]: I mean if the hub returns some other error other than the wrong timestamp error

with this change this happens:
 

[Tab-1676099282762.webm](https://user-images.githubusercontent.com/15967809/218245713-031e52d0-54be-42f5-a6bb-7c4c55b808dc.webm)

Comment (samuveth) [reply src/components/ModalVote.vue:90]: Right good catch! I will create another issue for this
